salt/utils/cloud.py

```diff
@@ -2262,6 +2289,13 @@ def sftp_file(dest_path, contents=None, kwargs=None, local_file=None):
         ssh_args = [
             # Don't add new hosts to the host key database
             "-oStrictHostKeyChecking=no",
+            # make sure ssh can time out on connection lose
+            "-oServerAliveInterval={}".format(
+                kwargs.get("server_alive_interval", SERVER_ALIVE_INTERVAL)
+            ),
+            "-oServerAliveCountMax={}".format(
+                kwargs.get("server_alive_count_max", SERVER_ALIVE_COUNT_MAX)
```

Comment:
I also noticed the function def root_cmd has ssh_args as well. Does that function also need to be updated to allow a user to pass in these two new kwargs?

Reply:
The options are only being added when `StrictHostKeyChecking` is present for this PR.

Reply:
`ssh_file` passes down the `new kwargs` via the `ssh_kwargs` from what I'm seeing and I don't see where I'm missing the kwarg passing in `wait_for_port.`